### ISSUE TITLE
audio pause

### ISSUE THREAD

**Issue by vonseg1** [iina/iina#4709]
**System and IINA version:**

- macOS  14.1.1
- IINA  1.3.3

**Expected behavior:**
Continuous sound when playing an audio file such as *.mp4 or *.m4a, with no pauses or interruptions.

**Actual behavior:**
IINA pauses for about 1 second at irregular intervals separated by a few minutes, 5 or less.  This has actually happened with every update of IINA I have made.  It happens even if no other application or task is taking up Mac CPU or memory.  

<details>
<summary><b>Crash report:</b></summary>

```
```
</details>

<br>

<details>
<summary><b>mpv log:</b></summary>

```
```
</details>

<br>

**Steps to reproduce:**

- [ ] MPV does not have this problem.

**How often does this happen?**
At least once per audio track, but irregularly.

**Comment by low-batt** [comment]
This report is concerning.

Open [Terminal](https://support.apple.com/guide/terminal/welcome/mac) and run this command:
```text
ps -e -O command,pid,start | grep -e 'coreaudiod\|STARTED' | grep -v grep
```
Capture the output. Then reproduce the problem. Once audio restarts run that command again. Post the before and after output here.

This is what running that command looks like on my Mac:
```text
low-batt@gag ~$ ps -e -O command,pid,start | grep -e 'coreaudiod\|STARTED' | grep -v grep
  PID COMMAND            PID STARTED   TT  STAT      TIME COMMAND
  195 /usr/sbin/coreau   195  3Nov23   ??  Ss     6:16.62 /usr/sbin/coreaudiod
low-batt@gag ~$ 
```

The [ps](https://ss64.com/osx/ps.html) command is displaying  information about processes running in memory. The output of that command is large, so we are using the [grep](https://ss64.com/bash/grep.html) to filter it down to the information we are interested in.

The output we want to see is that the time shown in the `STARTED` column does not change for `coreaudiod`. The output above shows `3Nov23`. That corresponds to when macOS was started:
```bash
low-batt@gag ~$ uptime
16:31  up 31 days, 23:15, 6 users, load averages: 3.62 2.67 2.62
low-batt@gag ~$ 
```

If the start time of `coreaudiod` does not match when macOS was started then that means this macOS daemon is restarting. We need to rule that out.

Also open Apple's [Console](https://support.apple.com/guide/console/welcome/mac) app and look for crash reports as explained [here](https://support.apple.com/guide/console/reports-cnsl664be99a/mac). If a `coreaudiod` crash is found, drag and drop the `.ips` file to this issue. You might have to zip the file to get GitHub to accept it.

**Comment by vonseg1** [comment]
What I get in response to your directions is below.  Does not seem like much help.  I run a fairly clean Mac O/S Sonoma system.  

David von Seggern


DavidsMini1758:Desktop david1$ ps -e -O command,pid,start | grep -e 'coreaudiod\|STARTED' | grep -v grep
  PID COMMAND            PID STARTED   TT  STAT      TIME COMMAND
  625 /usr/sbin/coreau   625 18Nov23   ??  Ss    31:44.77 /usr/sbin/coreaudiod

IINA audio pause happened here

DavidsMini1758:Desktop david1$ ps -e -O command,pid,start | grep -e 'coreaudiod\|STARTED' | grep -v grep
  PID COMMAND            PID STARTED   TT  STAT      TIME COMMAND
  625 /usr/sbin/coreau   625 18Nov23   ??  Ss    31:45.06 /usr/sbin/coreaudiod

I ran the uptime command about an hour later.
DavidsMini1758:Desktop david1$ uptime
20:50  up 17 days,  5:54, 2 users, load averages: 1.53 1.33 1.27

No coreaudiod crash reports in Console.



 

    On Tuesday, December 5, 2023 at 04:44:42 PM EST, low-batt ***@***.***> wrote:  
 
 


This report is concerning.

Open Terminal and run this command:
ps -e -O command,pid,start | grep -e 'coreaudiod\|STARTED' | grep -v grep

Capture the output. Then reproduce the problem. Once audio restarts run that command again. Post the before and after output here.

This is what running that command looks like on my Mac:
***@***.*** ~$ ps -e -O command,pid,start | grep -e 'coreaudiod\|STARTED' | grep -v grep
  PID COMMAND            PID STARTED   TT  STAT      TIME COMMAND
  195 /usr/sbin/coreau   195  3Nov23   ??  Ss     6:16.62 /usr/sbin/coreaudiod
***@***.*** ~$ 

The ps command is displaying information about processes running in memory. The output of that command is large, so we are using the grep to filter it down to the information we are interested in.

The output we want to see is that the time shown in the STARTED column does not change for coreaudiod. The output above shows 3Nov23. That corresponds to when macOS was started:
***@***.*** ~$ uptime
16:31  up 31 days, 23:15, 6 users, load averages: 3.62 2.67 2.62
***@***.*** ~$ 
If the start time of coreaudiod does not match when macOS was started then that means this macOS daemon is restarting. We need to rule that out.

Also open Apple's Console app and look for crash reports as explained here. If a coreaudiod crash is found, drag and drop the .ips file to this issue. You might have to zip the file to get GitHub to accept it.

**Comment by low-batt** [comment]
Thanks for running that test. A `coreaudiod` crash could cause a momentary problem with audio while that daemon restarts. The results indicates we can rule that out as the cause.

For audio/video support IINA uses a library from the [mpv](https://github.com/mpv-player/mpv) project. This is why the issue template contains this checkbox:
> - [ ] MPV does not have this problem.

If the problem is reproducible with the [mpv](https://mpv.io/) player then the root cause is in code IINA does not control and the problem must be reported to that project. IINA code is involved with certain aspects of video. That is not true for audio. The `mpv` library directly outputs audio. So it is more likely this problem may need to be reported to that group. But we aren't there yet. Lets see if there is some evidence of the problem in the log files.

To enable logging follow these instructions:
- Start IINA
- Click on  `Settings…` under the `IINA` menu
- The settings panel appears
- On the left side of the panel click on `Advanced`
- Slide the `Enable advanced settings` toggle button to be on (blue)
- Click on the checkbox `Enable logging to file` 
- Confirm the settings panel looks like the screenshot below
- **Restart IINA** to activate logging

![issue-4502-logging](https://github.com/iina/iina/assets/86170219/d60e23a0-76af-43ee-827c-b5483785f684)

Once IINA has restarted, play the audio file and reproduce the pause. Then:

- Click on  `Settings…` under the `IINA` menu
- The settings panel appears
- On the left side of the panel click on `Advanced`
- Click on `Open log directory`
- A new `Finder` window appears containing the logs for this session
- **Quit IINA** to complete the log files

The `Finder` window contains two files, `iina.log` and `mpv.log`. To attach the log files to this issue, drag-n-drop them from the `Finder` window to the `Leave a comment` section below. Include both the `iina.log` and the `mpv.log`

The log files can also be accessed using [Terminal](https://support.apple.com/guide/terminal/welcome/mac) as shown here:
```bash
low-batt@gag com.colliderli.iina$ pwd
/Users/low-batt/Library/Logs/com.colliderli.iina
low-batt@gag com.colliderli.iina$ ls
2022-10-04-17-57-15_jVrbWy
low-batt@gag com.colliderli.iina$ cd 2022-10-04-17-57-15_jVrbWy
low-batt@gag 2022-10-04-17-57-15_jVrbWy$ ls
iina.log	mpv.log
low-batt@gag 2022-10-04-17-57-15_jVrbWy$ 
```

With logging enabled IINA creates a new log directory each time it is started. If accessing the logs using `Terminal` be sure to obtain them from the latest log directory.

**DO NOT FORGET** to disable logging once you have obtained log files for the problem being investigated. IINA does not restrict the size of log files or delete old logs. If you leave logging enabled the logs will continue to accumulate.

**Comment by vonseg1** [comment]
Thanks again for your attention to my IINA issues.
I have attached the iina.log and mpv.log files to this email -- sorry, but I did not understand your instruction to "drag-n-drop them from the Finder window to the Leave a comment section below."
I note that the iina.log file has the entry:
11:43:20.829 [mpv][w] [cplayer] warn: Audio device underrun detected.

which coincides with the time I noted the audio pause of about 1 second.

Are we closing in on the problem?
David von Seggern

 

    On Tuesday, December 5, 2023 at 11:24:50 PM EST, low-batt ***@***.***> wrote:  
 
 


Thanks for running that test. A coreaudiod crash could cause a momentary problem with audio while that daemon restarts. The results indicates we can rule that out as the cause.

For audio/video support IINA uses a library from the mpv project. This is why the issue template contains this checkbox:

   
   -  MPV does not have this problem.


If the problem is reproducible with the mpv player then the root cause is in code IINA does not control and the problem must be reported to that project. IINA code is involved with certain aspects of video. That is not true for audio. The mpv library directly outputs audio. So it is more likely this problem may need to be reported to that group. But we aren't there yet. Lets see if there is some evidence of the problem in the log files.

To enable logging follow these instructions:
   
   - Start IINA
   - Click on Settings… under the IINA menu
   - The settings panel appears
   - On the left side of the panel click on Advanced
   - Slide the Enable advanced settings toggle button to be on (blue)
   - Click on the checkbox Enable logging to file
   - Confirm the settings panel looks like the screenshot below
   - Restart IINA to activate logging

issue-4502-logging.jpg (view on web)

Once IINA has restarted, play the audio file and reproduce the pause. Then:
   
   - Click on Settings… under the IINA menu
   - The settings panel appears
   - On the left side of the panel click on Advanced
   - Click on Open log directory
   - A new Finder window appears containing the logs for this session
   - Quit IINA to complete the log files

The Finder window contains two files, iina.log and mpv.log. To attach the log files to this issue, drag-n-drop them from the Finder window to the Leave a comment section below. Include both the iina.log and the mpv.log

The log files can also be accessed using Terminal as shown here:
***@***.*** com.colliderli.iina$ pwd
/Users/low-batt/Library/Logs/com.colliderli.iina
***@***.*** com.colliderli.iina$ ls
2022-10-04-17-57-15_jVrbWy
***@***.*** com.colliderli.iina$ cd 2022-10-04-17-57-15_jVrbWy
***@***.*** 2022-10-04-17-57-15_jVrbWy$ ls
iina.log	mpv.log
***@***.*** 2022-10-04-17-57-15_jVrbWy$ 
With logging enabled IINA creates a new log directory each time it is started. If accessing the logs using Terminal be sure to obtain them from the latest log directory.

DO NOT FORGET to disable logging once you have obtained log files for the problem being investigated. IINA does not restrict the size of log files or delete old logs. If you leave logging enabled the logs will continue to accumulate.

**Comment by low-batt** [comment]
I received your email, but apparently GitHub stripped the attachments.

Dragging and dropping is explained [here](https://support.apple.com/guide/mac-help/drag-and-drop-items-mh35852/mac). To add the log files to this issue bring up a `Finder` window that shows the two log files. Click on one of those files and hold down the button. While continuing to hold down the button move the mouse to the browser. The file will move with the cursor. This is referred to as dragging. Move the cursor over the browser inside of the `Add a comment` section below in the box that says `Add your comment here…`. Then release the mouse button. This "drops" the file into the browser. That will cause the browser to upload the file to GitHub. You will see special text appear in the box. Do not modify that text. If you then click on the `Preview` tab you will see a link with the filename of the file that can be used to download the file from GitHub. Do that for both the `iina.log` and `mpv.log` files. Then click on `Comment`.

The `Audio device underrun detected` log message is a symptom of the failure. I need to review the entire log file to see if there is some indication of what is triggering the problem.

**Comment by vonseg1** [comment]
[iina.log](https://github.com/iina/iina/files/13594526/iina.log)
[mpv.log](https://github.com/iina/iina/files/13594528/mpv.log)

**Comment by vonseg1** [comment]
I think I sent the files to GitHub.  You will, of course, know.  I am using this Issue interface for the first time.  I did not understand your instruction
"If you then click on the Preview tab you will see a link with the filename of the file that can be used to download the file from GitHub. Do that for both the iina.log and mpv.log files. 

Why should I download the two files?  I just uploaded them from my computer.
David von Seggern

 

    On Wednesday, December 6, 2023 at 09:25:50 PM EST, low-batt ***@***.***> wrote:  
 
 


I received your email, but apparently GitHub stripped the attachments.

Dragging and dropping is explained here. To add the log files to this issue bring up a Finder window that shows the two log files. Click on one of those files and hold down the button. While continuing to hold down the button move the mouse to the browser. The file will move with the cursor. This is referred to as dragging. Move the cursor over the browser inside of the Add a comment section below in the box that says Add your comment here…. Then release the mouse button. This "drops" the file into the browser. That will cause the browser to upload the file to GitHub. You will see special text appear in the box. Do not modify that text. If you then click on the Preview tab you will see a link with the filename of the file that can be used to download the file from GitHub. Do that for both the iina.log and mpv.log files. Then click on Comment.

The Audio device underrun detected log message is a symptom of the failure. I need to review the entire log file to see if there is some indication of what is triggering the problem.

**Comment by low-batt** [comment]
Sorry for the confusion. Bad wording on my part. The "Do that part" was about uploading the files, not downloading them. No need for you to download them.

The upload of `iina.log` worked, but the `mpv.log` is zero length. Can you please try uploading that file again.

**Comment by vonseg1** [comment]
I had to remake both log files.  Same problem occurs now at 23:55:56 in iina.log file.  mpv.log reflects this.  

[iina.log](https://github.com/iina/iina/files/13595357/iina.log)
[mpv.log](https://github.com/iina/iina/files/13595359/mpv.log)

**Comment by vonseg1** [comment]
I uploaded new log files.
Should I even be emailing you?  Seems like the GitHub issue reporting interface is all we need.  

David von Seggern 

    On Wednesday, December 6, 2023 at 11:24:15 PM EST, low-batt ***@***.***> wrote:  
 
 


Sorry for the confusion. Bad wording on my part. The "Do that part" was about uploading the files, not downloading them. No need for you to download them.

The upload of iina.log worked, but the mpv.log is zero length. Can you please try uploading that file again.

**Comment by low-batt** [comment]
Thanks for uploading the log files. On this:
> Should I even be emailing you?  Seems like the GitHub issue reporting interface is all we need.

Correct. No need to email me. Posting comments on GitHub is fine.

I see the audio file is on a mounted device. Is that some sort of network attached storage?

The log shows this:
```text
[ 115.699][w][cplayer] Audio device underrun detected.
[ 115.700][v][cplayer] Enter buffering (buffer went from 100% -> 0%) [0.000000s].
[ 116.644][v][cplayer] Still buffering (buffer went from 0% -> 16%) [0.162540s].
[ 116.644][v][cplayer] Still buffering (buffer went from 16% -> 34%) [0.348299s].
[ 116.645][v][cplayer] End buffering (waited 0.945053 secs) [1.021678s].
[ 116.645][v][cplayer] restarting audio after underrun
```

Two things are happening. For some reason `mpv` was temporarily unable to read bytes from the audio file. The delay was long enough that `mpv` used up all of the data it had cached and had nothing to play, so audio playback stalled.

At that point a `mpv` feature kicked in. From the `mpv` manual entry for the option [cache-pause](https://mpv.io/manual/stable/#options-cache-pause):
> **--cache-pause=<yes|no>**
>    Whether the player should automatically pause when the cache runs out of data and stalls decoding/playback (default: yes). If enabled, it will pause and unpause once more data is available, aka "buffering".

I guess the idea behind this feature is to try and cache some data before restarting playback inorder to try and avoid a series of short play, stop, play, stop cycles.

From [cache-pause-wait](https://mpv.io/manual/stable/#options-cache-pause-wait):
> **--cache-pause-wait=\<seconds>**
> Number of seconds the packet cache should have buffered before starting playback again if "buffering" was entered (default: 1).

Thus `mpv` is intentionally keeping audio playback paused while trying to cache a second of audio data before starting to play audio again.

Seems like something is intermittently causing reading of data from the audio file to wait. This could be a background task. Maybe something like [Spotlight](https://support.apple.com/guide/mac-help/search-with-spotlight-mchlp1008/mac) indexing.

If you move the audio file to the Mac's local disk does the problem reproduce?

As an experiment we can try and instruct `mpv` to buffer more data. The hope being that `mpv` will have cached enough audio data so that it does not run out of data during these moments when reading data from the device slows down. I believe this can be done by setting [demuxer-readahead-secs](https://mpv.io/manual/stable/#options-demuxer-readahead-secs):
> **--demuxer-readahead-secs=\<seconds>**
> If `--demuxer-thread` is enabled, this controls how much the demuxer should buffer ahead in seconds (default: 1).

To do that follow these instructions:
- Click on  `Settings…` under the `IINA` menu
- The settings panel appears
- On the left side of the panel click on `Advanced`
- Slide the `Enable advanced settings` toggle button to be on (blue)
- In the `Additional mpv options` section click on `+`
- A new entry appears in the table
- Double click on `name` and replace it with `demuxer-readahead-secs`
- Double click on `value` and replace it with `5`
- Confirm the settings panel looks like the screenshot below
- **Restart IINA** to activate the new settings

![issue-4709](https://github.com/iina/iina/assets/86170219/1fbf9491-3366-4ab0-8164-5b93e3c74273)

At least I think this will allow `mpv` to cache more of the audio data. See if this change alters the behavior at all.

**Comment by vonseg1** [comment]
I changed the value of demuxer-readahead-secs to 5.  I restarted IINA, but the pause problem persists -- no apparent change.  The log files are attached for this latest trial.
[iina.log](https://github.com/iina/iina/files/13615821/iina.log)
[mpv.log](https://github.com/iina/iina/files/13615824/mpv.log)

Another piece of information -- I am working on this issue from a desktop Mac Sonoma system.  This same pause problem occurs on my Mac laptop running the Monterey O/S and the latest IINA version.  I am puzzled why this problem is not endemic among IINA Mac users -- am I the only one noticing these pauses?

**Comment by vonseg1** [comment]
Henceforth I am using the GitHub issue interface.  I just left a comment.

David von Seggern 

    On Thursday, December 7, 2023 at 10:37:29 PM EST, low-batt ***@***.***> wrote:  
 
 


Thanks for uploading the log files. On this:


Should I even be emailing you? Seems like the GitHub issue reporting interface is all we need.


Correct. No need to email me. Posting comments on GitHub is fine.

I see the audio file is on a mounted device. Is that some sort of network attached storage?

The log shows this:
[ 115.699][w][cplayer] Audio device underrun detected.
[ 115.700][v][cplayer] Enter buffering (buffer went from 100% -> 0%) [0.000000s].
[ 116.644][v][cplayer] Still buffering (buffer went from 0% -> 16%) [0.162540s].
[ 116.644][v][cplayer] Still buffering (buffer went from 16% -> 34%) [0.348299s].
[ 116.645][v][cplayer] End buffering (waited 0.945053 secs) [1.021678s].
[ 116.645][v][cplayer] restarting audio after underrun

Two things are happening. For some reason mpv was temporarily unable to read bytes from the audio file. The delay was long enough that mpv used up all of the data it had cached and had nothing to play, so audio playback stalled.

At that point a mpv feature kicked in. From the mpv manual entry for the option cache-pause:


--cache-pause=<yes|no>
Whether the player should automatically pause when the cache runs out of data and stalls decoding/playback (default: yes). If enabled, it will pause and unpause once more data is available, aka "buffering".


I guess the idea behind this feature is to try and cache some data before restarting playback inorder to try and avoid a series of short play, stop, play, stop cycles.

From cache-pause-wait:


--cache-pause-wait=<seconds>
Number of seconds the packet cache should have buffered before starting playback again if "buffering" was entered (default: 1).


Thus mpv is intentionally keeping audio playback paused while trying to cache a second of audio data before starting to play audio again.

Seems like something is intermittently causing reading of data from the audio file to wait. This could be a background task. Maybe something like Spotlight indexing.

If you move the audio file to the Mac's local disk does the problem reproduce?

As an experiment we can try and instruct mpv to buffer more data. The hope being that mpv will have cached enough audio data so that it does not run out of data during these moments when reading data from the device slows down. I believe this can be done by setting demuxer-readahead-secs:


--demuxer-readahead-secs=<seconds>
If --demuxer-thread is enabled, this controls how much the demuxer should buffer ahead in seconds (default: 1).


To do that follow these instructions:
   
   - Click on Settings… under the IINA menu
   - The settings panel appears
   - On the left side of the panel click on Advanced
   - Slide the Enable advanced settings toggle button to be on (blue)
   - In the Additional mpv options section click on +
   - A new entry appears in the table
   - Double click on name and replace it with demuxer-readahead-secs
   - Double click on value and replace it with 5
   - Confirm the settings panel looks like the screenshot below
   - Restart IINA to activate the new settings

issue-4709.jpg (view on web)

At least I think this will allow mpv to cache more of the audio data. See if this change alters the behavior at all.

**Comment by low-batt** [comment]
I regret asking you to run that test as it resulted in my important question being missed:

What is this mount point attached to?: `/Volumes/music`

Is this cloud based storage? Local network attached storage? Is the connection to this going over Wi-Fi? Over the Internet?

On this:
>  I am puzzled why this problem is not endemic among IINA Mac users -- am I the only one noticing these pauses?

I am not aware of another issue that matches this. That is why my first response had you rule out `coreaudiod` problems on your Mac. _However_…

The log messages are symptomatic of `mpv` momentarily being unable to obtain data from the audio file. Computer programs that play audio/video are considered [soft real-time](https://en.wikipedia.org/wiki/Real-time_computing#Criteria_for_real-time_computing) systems. They require a steady stream of data from the media being played. Typically such programs attempt to "read ahead" and cache data in order for them to be able to tolerate some delay in obtaining the next chunk of data. But if that delay is too long then the player runs out of data and has no choice but to stop playback.

You will see this behavior with IINA when playing videos from a streaming site. This can occur because:
- The streaming site is overloaded
- The streaming site has intentionally throttled bandwidth to your IP
- Your ISP is experiencing congestion
- Your neighbor in the apartment next door started an appliance that is interfering with your Wi-Fi

In other words anything that slows down the stream of data too much from the media's location to your Mac.

If you look through IINA's issues you will find reports of pausing and buffering while streaming YouTube videos. YouTube is known to sometimes throttle the bandwidth it permits a client to use. This is especially noticeable when streaming videos at high resolutions.

Thus my question about what `/Volumes/music` is connected to. We want to know if reading data from that mount point is constrained and sometimes stalling.

**Comment by vonseg1** [comment]
This issue is closed.  The variable demuxer-readahead-secs was apparently NOT set to a value of 5 for some unknown reason (maybe my typo).  I found it blank in the Advanced section of Preferences.  So I again set it to 5, closed IINa, restarted IINA, checked the settings -- demuxer-readahead-secs is now 5.  I played about 15 minutes of audio files -- no pauses were detected.  My audio files are on an external disk connected by USB 3 cable.  Apparently demuxer-readahead-secs set to 1 in this case is not sufficient to maintain a steady stream of audio to my speaker.  I appreciate the help given by "low-batt", as I am a first-timer to this GitHub issue interface.

**Comment by low-batt** [comment]
Glad the `demuxer-readahead-secs` workaround is working for you. Thank you for reporting back and closing the issue.

I am really surprised and troubled this is happening with an external USB disk, especially with playing audio which does not need the bandwidth required for video.

Is this external disk a hard disk drive or a solid state drive? If this is a HDD then it is important confirm this is not because the drive is starting to fail. From [How To Successfully Repair Bad Sectors on Mac External Hard Drive](https://www.macobserver.com/tips/how-to/successfully-repair-bad-sectors-on-mac-external-hard-drive/):
>  **Signs of Bad Sectors on Your Mac External HDD**
>
>    Slow performance: When your Mac reads data from an external hard disk drive with bad sectors, it has to skip over those sectors. This slows down the reading process and can make the Mac feel sluggish and unresponsive.

When a hard disk starts failing it has to read bad spots multiple times, slowing the time to read data. Everything is working, just slow. Not clear that is happening in this case. I'm only mentioning this because when this happens it is important to back up your data as the situation can deteriorate quickly.

The macOS [Disk Utility](https://support.apple.com/guide/disk-utility/welcome/mac) can be used to check the disk.

USB is known for problems. I recently read about a `mpv` issue which was traced to a USB printer which was interfering with other USB devices. Keep that in mind as well.

**Comment by vonseg1** [comment]
Thanks for the final advice.  Your attention to my issue has been above and beyond the call of duty it seems to me.  

David von Seggern


 

    On Friday, December 8, 2023 at 07:18:34 PM EST, low-batt ***@***.***> wrote:  
 
 


Glad the demuxer-readahead-secs workaround is working for you. Thank you for reporting back and closing the issue.

I am really surprised and troubled this is happening with an external USB disk, especially with playing audio which does not need the bandwidth required for video.

Is this external disk a hard disk drive or a solid state drive? If this is a HDD then it is important confirm this is not because the drive is starting to fail. From How To Successfully Repair Bad Sectors on Mac External Hard Drive:


Signs of Bad Sectors on Your Mac External HDD

Slow performance: When your Mac reads data from an external hard disk drive with bad sectors, it has to skip over those sectors. This slows down the reading process and can make the Mac feel sluggish and unresponsive.


When a hard disk starts failing it has to read bad spots multiple times, slowing the time to read data. Everything is working, just slow. Not clear that is happening in this case. I'm only mentioning this because when this happens it is important to back up your data as the situation can deteriorate quickly.

The macOS Disk Utility can be used to check the disk.

USB is known for problems. I recently read about a mpv issue which was traced to a USB printer which was interfering with other USB devices. Keep that in mind as well.

**Comment by low-batt** [comment]
Your very welcome. Should the workaround not prove fully effective do not hesitate to reopen this issue or enter a new one.

That this involved a locally attached USB disk caught me by surprise. I'm still digesting that. I will be on the lookout for any internal IINA/`mpv` issues that could be a contributing factor.
